diff --git a/pom.xml b/pom.xml
index 596c980..bc1b286 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
 
   <modelVersion>4.0.0</modelVersion>
 
diff --git a/src/main/java/com/obones/binding/openmeteo/internal/discovery/OpenMeteoDiscoveryService.java b/src/main/java/com/obones/binding/openmeteo/internal/discovery/OpenMeteoDiscoveryService.java
index 51fc56f..def67e2 100644
--- a/src/main/java/com/obones/binding/openmeteo/internal/discovery/OpenMeteoDiscoveryService.java
+++ b/src/main/java/com/obones/binding/openmeteo/internal/discovery/OpenMeteoDiscoveryService.java
@@ -78,7 +78,7 @@ public class OpenMeteoDiscoveryService extends AbstractDiscoveryService {
     @Override
     public void deactivate() {
         logger.debug("Removing older discovery services.");
-        removeOlderResults(Instant.now(), bridgeHandler.getThing().getUID());
+        removeOlderResults(Instant.now().toEpochMilli(), bridgeHandler.getThing().getUID());
         super.deactivate();
     }
 
